soc/nxp/mcx/mcxw/Kconfig.defconfig

```diff
@@ -12,4 +12,23 @@ config SYS_CLOCK_HW_CYCLES_PER_SEC
 config MCUX_FLASH_K4_API
 	default y
 
+if BT
+
+# Include intercore messaging component
+config NXP_RF_IMU
+	default y
+
+#  Set the controller's public identity using NXP vendor command
+config BT_HCI_SET_PUBLIC_ADDR
+	default y
+
+# HCI RX buffers are received in ISR context. RX messages
+# need to be queued and processed by a dedicated thread
+config HCI_NXP_RX_THREAD
+	default y
+
+config HEAP_MEM_POOL_SIZE
+	default 1024
```

Comment:
I'm not sure that this belongs here, maybe you meant to put it in the frdm_mcxw71_defconfig. I also just noticed it's being defaulted in the driver kconfig too, that shouldn't happen either. This is an app specific configuration, the closest reasonable place we have to put it is the board_defconfig, which is what has been done in the past, you can see in tree no driver or soc defaults this, but some boards set it.

Reply:
On RW we initially defaulted this in the board defconfig, and it was moved at driver level afterwards (I think you did when refactoring kconfig stuff some time ago).
@yeaissa found `HEAD_MEM_POOL_ADD_SIZE_*` kconfig which allows for subsystem/drivers to declare their minimal heap needs. Then the heap size is guaranteed to be at least the sum of all those values. We'll update the PR using this, seems like the best approach.

Reply:
I removed `HEAP_MEM_POOL_SIZE` and replaced with `HEAD_MEM_POOL_ADD_SIZE_BT_NXP` and `HEAD_MEM_POOL_ADD_SIZE_BT_NXP_THREAD` which indicates the amount of heap needed by this driver.

Reply:
> On RW we initially defaulted this in the board defconfig, and it was moved at driver level afterwards (I think you did when refactoring kconfig stuff some time ago)

It was in the SOC file before and I moved it to the driver it looks like when I was mass moving things out of the soc level, which is slightly better I guess but it really should have been moved to the board _defconfig instead of the driver, that much was my mistake, but we shouldn't add it back to the soc level

> I removed HEAP_MEM_POOL_SIZE and replaced with HEAD_MEM_POOL_ADD_SIZE_BT_NXP and HEAD_MEM_POOL_ADD_SIZE_BT_NXP_THREAD which indicates the amount of heap needed by this driver.

I'm confused why, because I don't see what they are being used for?

Reply:
> I removed `HEAP_MEM_POOL_SIZE` and replaced with `HEAP_MEM_POOL_ADD_SIZE_BT_NXP` and `HEAP_MEM_POOL_ADD_SIZE_BT_NXP_RX_THREAD` which indicates the amount of heap needed by this driver.

this is very interesting actually, I didn't know about this, it's very useful, thanks